### DISCUSSION TITLE
GITHUB_HEAD_REF is inconsistent

### DISCUSSION THREAD

**Discussion by SamuraiTheNinja** [community/community#25773] [Actions]
<p>Opposite from this topic: <a href="https://github.community/t/github-ref-is-inconsistent/17728" class="inline-onebox">GITHUB_REF is inconsistent</a><br>
I need the name of the branch from within my GitHub actions.</p>
<h2>
<a name="issue-1" class="anchor" href="#issue-1"></a>Issue:</h2>
<ul>
<li>On pull_request event → expected outcome, I can use the environment variable github.head_ref to get the name of the branch since github.ref is a pull request</li>
<li>On pull_request_review event → github.head_ref is empty, but github.ref still reflects the pull request, leaving me without the branch name in this case.</li>
</ul>
<h2>
<a name="workflow-that-requires-this-2" class="anchor" href="#workflow-that-requires-this-2"></a>Workflow that requires this:</h2>
<p>There are branch protection rules in place on master to prevent anyone (including the GitHub action as we can’t exclude it) from pushing to master. So the file that gets changed within the action needs to be pushed to the pull request incoming branch. We also require pull requests be approved by a reviewer before they can be merged. So we do not want this to fire on a pull_request, but only on the pull_request_review (since no pull_request_approved event exists)</p>
<p>I verified this behavior by echoing both github.head_ref and github.ref for a pull_request event, and in a separate pull request for the event pull_request_review.</p>
<p>Any help is appreciated</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/samuraitheninja">@SamuraiTheNinja</a>, welcome to the GitHub Support Community! Is it the head ref you’re looking for? I think you’ll be able to get this from inside the <code>github.event</code> context at <code>github.event.pull_request.head.ref</code> for both of these.</p>
<p>There’s example payloads for the webhook events that trigger workflows on <a href="https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request">this page</a> so I’d recommend double checking that this is what you’re looking for.</p>

**Comment by SamuraiTheNinja** [comment]
<p>Thank you <a class="mention" href="/thomasshaped">@thomasshaped</a> that has solved the problem I was having. Finding the right documentation for most of GitHub Actions has been a nightmare, so I really appreciate the help!</p>

**Comment by hafizmuhammadshoaib** [comment]
<p>Hey can you guys help me with the following thing</p>
<pre><code class="lang-auto">on:
  pull_request:
    branches: [develop, staging, qa, delivery-ops-qa]
jobs:
  dev_job:
    if: ${{ github.event.pull_request.head.ref == 'refs/heads/develop' }}
    runs-on: ubuntu-latest
    steps:
      - name: Configure DEV AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_DEV }}
          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_DEV }}
          aws-region: ap-southeast-1
      - name: Run CodeBuild
        uses: aws-actions/aws-codebuild-run-build@v1
        with:
          project-name: dev-delivery-ops-api-prebuild
  stg_job:
    runs-on: ubuntu-latest
    if:  ${{ github.event.pull_request.head.ref == 'refs/heads/staging' }}
    steps:
      - name: Configure STAGING AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_STAGING }}
          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_STAGING }}
          aws-region: ap-southeast-1
      - name: Run CodeBuild
        uses: aws-actions/aws-codebuild-run-build@v1
        with:
          project-name: staging-delivery-ops-api-prebuild
  qa_job:
    runs-on: ubuntu-latest
    if:  ${{ github.event.pull_request.head.ref == 'refs/heads/qa' || github.event.pull_request.head.ref == 'refs/heads/delivery-ops-qa'}}
    steps:
      - name: Configure QA AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_QA }}
          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_QA }}
          aws-region: ap-southeast-1
      - name: Run CodeBuild
        uses: aws-actions/aws-codebuild-run-build@v1
        with:
          project-name: qa-delivery-ops-api-prebuild
</code></pre>
<p>the problem is that none of my job triggers when I create a pull request against the following branches posted in my code. can you guys please help me with this</p>
<p>CC: <a class="mention" href="/thomasshaped1">@ThomasShaped1</a></p>
